### ISSUE TITLE
Cannot find service module info file

### ISSUE THREAD

**Issue by TanayParikh** [dotnet/aspnetcore#23333]
No consistent repro yet, so just putting this up as a tracking issue.

![image](https://user-images.githubusercontent.com/14852843/85634765-10384f80-b631-11ea-81b6-9682e895e885.png)

```
StreamJsonRpc.RemoteInvocationException : Cannot find service module info file 'razorLanguageService.servicehub.service.json' in 'C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\common7\ServiceHub\Services' (ErrorKind: ServiceModuleInfoNotFound)
   at async StreamJsonRpc.JsonRpc.InvokeCoreAsync[TResult](<Unknown Parameters>)
   at async Microsoft.ServiceHub.Client.HubClient.RemoteServiceBrokerWrapper.<>c__DisplayClass9_0.<RequestServiceChannelAsync>b__0(<Unknown Parameters>)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at async Microsoft.ServiceHub.Client.HubClient.RemoteServiceBrokerWrapper.<>c__DisplayClass9_0.<RequestServiceChannelAsync>b__0(<Unknown Parameters>)
   at async Microsoft.ServiceHub.Utility.Shared.ServiceHubRetry.ExecuteAsync[TReturnType](<Unknown Parameters>)
   at async Microsoft.ServiceHub.Utility.Shared.ServiceHubRetry.ExecuteAsync[TReturnType](<Unknown Parameters>)
   at async Microsoft.ServiceHub.Client.HubClient.RemoteServiceBrokerWrapper.RequestServiceChannelAsync(<Unknown Parameters>)
   at async Microsoft.ServiceHub.Framework.RemoteServiceBroker.GetPipeAsync(<Unknown Parameters>)
   at async Microsoft.ServiceHub.Client.HubClient.GetPipeAsync(<Unknown Parameters>)
   at async Microsoft.ServiceHub.Client.HubClient.RequestServiceAsync(<Unknown Parameters>)
   at async Microsoft.VisualStudio.LanguageServices.Remote.ServiceHubRemoteHostClient.RequestServiceAsync(<Unknown Parameters>)
---> (Remote Exception) {
  "type": "Microsoft.ServiceHub.Utility.ServiceHubError",
  "message": "Cannot find service module info file 'razorLanguageService.servicehub.service.json' in 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Preview\\common7\\ServiceHub\\Services' (ErrorKind: ServiceModuleInfoNotFound)",
  "stack": "   at Microsoft.ServiceHub.Controller.ServiceDiscovery.<GetServiceModuleInfoAsync>d__3.MoveNext()\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\r\n   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   at Microsoft.ServiceHub.Controller.LocationService.<GetHostedServiceLocationAsync>d__75.MoveNext()\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\r\n   at Microsoft.ServiceHub.Controller.LocationService.<LocateAsync>d__42.MoveNext()\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\r\n   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   at Microsoft.ServiceHub.Controller.HubController.<LocateAsync>d__8.MoveNext()\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\r\n   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   at Microsoft.ServiceHub.Controller.HubController.<RequestServiceChannelAsync>d__13.MoveNext()\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\r\n   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   at Microsoft.ServiceHub.Controller.ClientConnection.<RequestServiceChannelAsync>d__26.MoveNext()",
  "code": -2146233088,
  "inner": null
} <--- 

```

**Comment by NTaylorMullen** [comment]
@tmat have you seen this?

**Comment by tmat** [comment]
It reminds me a problem we had in Roslyn where sometimes the extension manager does not find the .json file in the extension dir. We worked around it by including it also under `ServiceHub\Services`. Would be nice if the platform fixed this issue.

```
c:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\Common7\IDE\CommonExtensions\Microsoft\ManagedLanguages\VBCSharp\LanguageServices
c:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\Common7\ServiceHub\Services\RoslynCodeAnalysisService
```

**Comment by NTaylorMullen** [comment]
> It reminds me a problem we had in Roslyn where sometimes the extension manager does not find the .json file in the extension dir. We worked around it by including it also under `ServiceHub\Services`. Would be nice if the platform fixed this issue.

Oh wow, ya that'd definitely be nice if the platform would fix the issue.... How are you guys able to contribute your json files to that servicehub directory? Are you able to do it via your VSIX or do you insert it directly into VS or some other repo?

**Comment by TanayParikh** [comment]
The only repro I've been able to get for this is:

1. Launch an experimental instance
2. Make some change, build solution, try launching a new experimental instance while #1 is still running.

This leads to access issues (due to which I suspect the Razor extension fails to install). 

3. Close all experimental instances, and relaunch with `CTRL+F5`. 

This will usually result in the service module exception shown in this issue. Closing the experimental instance, and rebuilding solution resolves the issue.

Consequently, this seems to be a development specific error that shouldn't(?) be customer impacting. Thoughts on closing this out? @NTaylorMullen
